std/utf.d

```diff
@@ -1627,12 +1752,47 @@ version(unittest) private void testDecodeFront(R)(ref R range,
     }
 }
 
-version(unittest) private void testBothDecode(R)(R range,
+version(unittest) private void testDecodeBack(R)(ref R range,
                                                  dchar expectedChar,
-                                                 size_t expectedIndex,
+                                                 size_t expectedNumCodeUnits,
                                                  size_t line = __LINE__)
 {
+    static if (!isSomeString!R || !isBidirectionalRange!R || (isRandomAccessRange!R && !hasLength!R))
+        return;
```

Comment:
Suspicious - instead of silently ignoring such ranges, there simply shouldn't be calling this function.

Just delete this static if branch.